pulp_gem/app/tasks/synchronizing.py

```diff
@@ -72,54 +73,86 @@ async def run(self):
         """
         # Interpret policy to download Artifacts or not
         deferred_download = self.remote.policy != Remote.IMMEDIATE
+        remote_url = self.remote.url
+
+        # Read filters from remote
+        includes = self.remote.includes
+        excludes = self.remote.excludes
+        prereleases = self.remote.prereleases
 
         async with ProgressReport(
             message="Downloading versions list", total=1
         ) as pr_download_versions:
-            versions_url = urljoin(self.remote.url, "versions")
+            versions_url = urljoin(remote_url, "versions")
             versions_downloader = self.remote.get_downloader(url=versions_url)
-            versions_result = await versions_downloader.run()
+            try:
+                versions_result = await versions_downloader.run()
+            except ClientConnectionError as e:
+                raise Exception(f"Could not connect to host {e.host}")
             await pr_download_versions.aincrement()
 
         async with ProgressReport(message="Parsing versions list") as pr_parse_versions:
             async with ProgressReport(message="Parsing versions info") as pr_parse_info:
```

Comment:
Not a big deal if you don't want to change it this PR, but can we remove the double indent by putting the second `with` statement in the first one?

Reply:
I tried, but couldn't make it work with async.